api/app/src/main/kotlin/packit/service/PacketGroupService.kt

```diff
@@ -2,25 +2,42 @@ package packit.service
 
 import org.springframework.data.domain.Page
 import org.springframework.data.domain.Sort
+import org.springframework.http.HttpStatus
 import org.springframework.stereotype.Service
+import packit.exceptions.PackitException
 import packit.helpers.PagingHelper
 import packit.model.PacketGroup
 import packit.model.PageablePayload
+import packit.model.dto.PacketGroupDisplay
+import packit.model.dto.PacketGroupDisplayImpl
 import packit.repository.PacketGroupRepository
 
 interface PacketGroupService
 {
     fun getPacketGroups(pageablePayload: PageablePayload, filteredName: String): Page<PacketGroup>
+    fun getPacketGroupDisplay(name: String): PacketGroupDisplay
 }
 
 @Service
 class BasePacketGroupService(
-    private val packetGroupRepository: PacketGroupRepository
+    private val packetGroupRepository: PacketGroupRepository,
+    private val packetService: PacketService
 ) : PacketGroupService
 {
     override fun getPacketGroups(pageablePayload: PageablePayload, filteredName: String): Page<PacketGroup>
     {
         val packetGroups = packetGroupRepository.findAllByNameContaining(filteredName, Sort.by("name"))
         return PagingHelper.convertListToPage(packetGroups, pageablePayload)
     }
+
+    override fun getPacketGroupDisplay(name: String): PacketGroupDisplay{
+        val latestPacketId = packetGroupRepository.findLatestPacketIdForGroup(name)?.id
+        val packetGroup = packetGroupRepository.findByName(name)
+        if (latestPacketId == null || packetGroup == null) {
+            throw PackitException("doesNotExist", HttpStatus.NOT_FOUND)
+        }
+        val packetOrderlyMetadata = packetService.getMetadataBy(latestPacketId).custom?.get("orderly") as? Map<*, *>
+        val packetLongDescription = (packetOrderlyMetadata?.get("description") as? Map<*, *>)?.get("long") as? String
+        return PacketGroupDisplayImpl(packetGroup.latestDisplayName, packetLongDescription)
+    }
```

Comment:
I'm a bit confused by what's going on here- it's going to outpack server to fetch metadata at the time when the endpoint is called? I hadnt imagined that would happen, but rather that during `importPackets`. it would check for any new packet display names and update the packet group table accordingly - and then we'd return that display information when requested by front end. 

Oh... it looks like you are doing that. So this get from outpack server is just for descriptions? Why not just put those in the packet group table too?

Reply:
I guess I thought display names would be much more commonly referred to than descriptions, justifying the db column, but in practice that turned out not to be true.